src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,296 @@
+        (= color "danger-50")  colors/danger-60
+        (= color "success-50") colors/success-60)
+      :else
+      (cond
+        (= color "neutral-5")  colors/neutral-5
+        (= color "neutral-10") colors/neutral-10
+        (= color "neutral-40") colors/neutral-40
+        (= color "neutral-50") colors/neutral-50
+        (= color "danger-50")  colors/danger-50
+        (= color "success-50") colors/success-50))))
+
+(defn load-icon
+  [icon color]
+  [rn/view {:style style/icon}
+   [quo2/icon icon
+    {:color color}]])
+
+(def total-box 85)
+
+(def app-state (reagent/atom {:counter 0}))
+(def interval-id (reagent/atom nil)) ; To store the interval ID
+
+(defn stop-interval
+  []
+  (when @interval-id
+    (js/clearInterval @interval-id)
+    (reset! interval-id nil))) ; Clear the interval ID
+
+(defn clear-counter
+  []
+  (swap! app-state assoc :counter 0))
+
+(defn update-counter
+  [networkState]
+  (let [new-counter-value (-> @app-state :counter inc)]
+    (if (or (and (= networkState "pending") (> new-counter-value 1))
+            (and (= networkState "sending") (> new-counter-value 2))
+            (and (= networkState "confirmed") (> new-counter-value 4))
+            (and (= networkState "finalising") (> new-counter-value 18))
+            (and (= networkState "finalised") (> new-counter-value total-box))
+            (and (= networkState "error") (> new-counter-value 2)))
+      (stop-interval)
+      (swap! app-state assoc :counter new-counter-value))))
+
+(defn start-interval
+  [networkState]
+  (reset! interval-id
+    (js/setInterval
+     (fn []
+       (update-counter networkState))
+     50))) ; Interval of 50ms
+
+(defn calculate-box-state
+  [networkState counter index]
+  (cond
+    (and (= networkState "sending") (>= counter index) (< index 3))                 "confirmed"
+    (and (= networkState "confirmed") (>= counter index) (< index 5))               "confirmed"
+    (and (= networkState "finalising") (>= counter index) (< index 5))              "confirmed"
+    (and (= networkState "finalising") (>= counter index) (> index 4) (< index 20)) "finalised"
+    (and (= networkState "finalised") (>= counter index) (< index 5))               "confirmed"
+    (and (= networkState "finalised") (>= counter index) (> index 4))               "finalised"
+    (and (= networkState "error") (>= counter index) (< index 2))                   "error"
+    :else                                                                           "pending"))
+
+(defn progress-boxes
+  [networkState]
+  [rn/view
+   {:style style/progress-box-container}
+   (let [numbers (range 1 total-box)]
+     (doall (for [n numbers]
+              [progress-box/progress-bar
+               {:network-state    (calculate-box-state networkState (@app-state :counter) n)
+                :width            "8"
+                :height           "12"
+                :marginHorizontal 2
+                :key              n
+               }])))])
```

Comment:
Based on Figma, The `state` property of the `Progress bar` has only four states `pending`, `confirmed`, `finalized`, and `error`.  

The `Transaction progress` component has six states: `pending`, `sending`, `confirmed`, `finalising`, `finalized`, and `error`. Please avoid introducing parent component props to children.

Reply:
Have updated this & yes, I am only sending those 4 states in the `Progress Bar` using this fn `calculate-box-state`, plz correct me if I am wrong @smohamedjavid